doc/ControlProtocol/ControlProtocol.tex

```diff
@@ -0,0 +1,886 @@
+        \hline
+        INVALID\_STATE & 0x01 & The requested state change was for an unknown state \\
+        \hline
+        PERMISSION\_DENIED & 0x02 & The sending entity does not have permission to update 
+        the state or configuration of the receiving entity. \\
+        \hline
+    \end{tabular}
+\end{table}
+
+\subsection{DEBUG Packets}
+\begin{table}[h!]
+    \centering
+    \label{tab:debugDef}
+    \caption{Structure of DEBUG packets}
+    \begin{tabular}{|p{1cm}|p{0.04cm}|p{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|
+        m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|
+        m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|}
+        \hline
+        Word & 
+        \begin{sideways}31\end{sideways} &
+        \begin{sideways}30\end{sideways} & 
+        \begin{sideways}29\end{sideways} &
+        \begin{sideways}28\end{sideways} &
+        \begin{sideways}27\end{sideways} &
+        \begin{sideways}26\end{sideways} &
+        \begin{sideways}25\end{sideways} &
+        \begin{sideways}24\end{sideways} &
+        \begin{sideways}23\end{sideways} &
+        \begin{sideways}22\end{sideways} &
+        \begin{sideways}21\end{sideways} &
+        \begin{sideways}20\end{sideways} &
+        \begin{sideways}19\end{sideways} &
+        \begin{sideways}18\end{sideways} &
+        \begin{sideways}17\end{sideways} &
+        \begin{sideways}16\end{sideways} &
+        \begin{sideways}15\end{sideways} &
+        \begin{sideways}14\end{sideways} &
+        \begin{sideways}13\end{sideways} &
+        \begin{sideways}12\end{sideways} &
+        \begin{sideways}11\end{sideways} &
+        \begin{sideways}10\end{sideways} &
+        \begin{sideways}9\end{sideways} &
+        \begin{sideways}8\end{sideways} &
+        \begin{sideways}7\end{sideways} &
+        \begin{sideways}6\end{sideways} &
+        \begin{sideways}5\end{sideways} &
+        \begin{sideways}4\end{sideways} &
+        \begin{sideways}3\end{sideways} &
+        \begin{sideways}2\end{sideways} &
+        \begin{sideways}1\end{sideways} &
+        \begin{sideways}0\end{sideways} \\
+        \hline
+        [0,1] & \multicolumn{32}{c|}{Header}\\
+        \hline
+        2 & \multicolumn{32}{c|}{Git Commit Hash}\\
+        \hline
+        3 & \multicolumn{32}{c|}{Robot UUID}\\
+        \hline
+        4 & \multicolumn{8}{c|}{Robot State} & \multicolumn{8}{c|}{\cellcolor[gray]{.6}Reserved}&
+            \multicolumn{16}{c|}{Number of config Entries}\\
+        \hline
+        ... & \multicolumn{32}{c|}{Arbitrary Data}\\
+        \hline
+        n & \multicolumn{16}{c|}{Arbitrary Data (Cont)} & \multicolumn{16}{c|}{\cellcolor[gray]{.6}Unused}\\
+        \hline
+    \end{tabular}
+\end{table}
+Git Commit Hash: The hash of the current git commit of the FMS or Robot code (not core code) running on the sender.
```

Comment:
What if this is a development version? 

`git describe --always` will git you a 28-bit short hash, 4 bits could be used to identify the status of the commit (i.e. on master, on dev branch, ahead of dev branch, etc.)

Reply:
Added